src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -201,10 +201,11 @@ public function channel(): \AMQPChannel
             $connection = $this->amqpFactory->createConnection($this->connectionCredentials);
             $connectMethod = 'true' === ($this->connectionCredentials['persistent'] ?? 'false') ? 'pconnect' : 'connect';
 
-            if (false === $connection->{$connectMethod}()) {
-                throw new \AMQPException('Could not connect to the AMQP server. Please verify the provided DSN.');
+            try {
+                $connection->{$connectMethod}();
+            } catch (\AMQPConnectionException $e) {
+                throw new \AMQPException(sprintf('Could not connect to the AMQP server. Please verify the provided DSN. (%s)', json_encode($this->connectionCredentials)), 0, $e);
```

Comment:
what about turning the password into `*****`?

Reply:
IIRC, the password is already in the trace

Reply:
Even so, I think only this top message would make it into the logs - seems like a good idea.

Reply:
Good catch. I will do that

Reply:
Fixed :+1: